samples/sqltransport-nhpersistence-outbox-ef/sample.md

```diff
@@ -70,4 +76,51 @@ Finally the messages in the outbox are pushed to their destinations. The timeout
+### Database connection sharing
+
+All the data manipulations happen atomically because SQL Server 2008 and later allows multiple (but not overlapping) instances of `SqlConnection` to enlist in one `TransactionScope` without the need to escalate to DTC. The SQL Server manages these transactions like they were one `SqlTransaction`. This does not work for all databases. This implementation shares the same `IDbConnection` between the dependencies that require the connection resulting on all dependencies to read and write to the same database.
+
+### NHibernateStorageContext
+
+NServiceBus.NHibernate has an `NHibernateStorageContext` which has a `IDbConnection` when it is available. This is used internally to share the connection to all persisters.
+
+We configure the container how it can obtain an `IDbConnection` and specify that this instance needs to be shared within the unit of work.
+
+<!-- import NHibernateStorageContextConnection -->
+
+
+### Custom DbContext pipeline behavior
+
+We cannot use `IManageUnitsOfWork` as this is to early in the pipeline as `NHibernateStorageContext` is initialized after the pipeline behavior responsible for managing `IManageUnitsOfWork`, so the connection to share is not yet available. 
+
+### Sharing the `IDbConnection`
+
+To get an initialized connection we create a custom pipeline behavior  which creates a configured `DbContext` via its generic argument.
+
+<!-- import DbContextBehavior -->
+
+### Behavior implementation
+
+When the behavior is called it simply create the `DbContext` and when the pipeline is finished it calls `DbContext.SaveChanges()`.
+
+<!-- import DbContextBehaviorSaveChanges -->
+
+The behavior itself is actually easier to implement that an `IManageUnitsOfWork`. The only drawback is that it needs additional code to register this behavior in the pipeline.
+
+### Configure bus to include `DbContext` behavior
+
+The only thing left is getting the behavior hooked in the pipeline. This is done by configuring the bus in a `INeedInitialization` created specific for our dbcontext behavior as we need to configure it to use `ReceiverDataContext` but also need to configure the container how it can resolve both the `DbContextBehavior<ReceiverDataContext>` behavior and `ReceiverDataContext` DbContext.
+
+<!-- import DbContextBehaviorContainerRegistration -->
+
+Register behavior in pipeline:
+
+<!-- import DbContextBehaviorPipelineRegistration -->
+
+## Alternatives
+
+An alternative of sharing the `IDbConnection` instance is instructing the container how to create the `ReceiverDataContext` by first resolving the `NHibernateStorageContext` and use that to build it.
+
+<!-- import ReceiverDataContextAlternative -->
+
+The current implementation is usefull when you would like to use the `IDbConnection` with, for example, the [Dapper](https://github.com/StackExchange/dapper-dot-net) micro orm framework.
```

Comment:
Does the comment about injecting an `IDbConnection` add value? I added the dapper remark to having more then one persistence framework (ef, nh, dapper, etc) that share the same connection.